org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/typesystem/CollectionInferenceTestCase.xtend

```diff
@@ -23,7 +23,7 @@ class CollectionInferenceTestCase extends AbstractWollokTypeSystemTestCase {
 			const l = []
 		}
 		'''.parseAndInfer.asserting [
-			assertTypeOf(classTypeFor(LIST), "l")
+			assertTypeOfAsString("List", "l")
```

Comment:
The method classTypeFor from the test cases, can not provide correct types when they are generic. You should ask the type system for it, maybe it could be fixed but